src/main/java/org/eln2/mc/utility/DataLabelBuilder.kt

```diff
@@ -0,0 +1,59 @@
+package org.eln2.mc.utility
+
+import kotlin.math.abs
+
+class DataLabelBuilder {
+    open class Entry(open val label : String, open val value : String, open val color : McColor)
+
+    private class SiEntry(override val label : String, val number : Double, val decimals : Int, val unit : String, override val color : McColor)
+        : Entry(label, SuffixConverter.convert(abs(number), unit, decimals), color)
+
+    val entries = ArrayList<Entry>()
+
+    fun entry(label : String, value : String, color : McColor) : DataLabelBuilder{
+        entries.add(Entry(label, value, color))
+        return this
+    }
+
+    fun entry(entry: Entry) : DataLabelBuilder{
+        entries.add(entry)
+        return this
+    }
+
+    fun siEntry(label: String, suffix : String, value: Double, color : McColor, decimals : Int = 2) : DataLabelBuilder{
+        return entry(SiEntry(label, value, decimals, suffix,  color))
+    }
+
+    fun volts(voltage : Double, color : McColor) : DataLabelBuilder {
+        return entry(SiEntry("Voltage", voltage, 2, "V", color))
+    }
+
+    fun ohms(resistance : Double, color : McColor) : DataLabelBuilder {
+        return entry(SiEntry("Resistance", resistance, 2, "Ω", color))
+    }
+
+    fun amps(current : Double, color : McColor) : DataLabelBuilder {
+        return entry(SiEntry("Current", current, 2, "A", color))
+    }
+
+    fun joules(energy : Double, color : McColor) : DataLabelBuilder {
+        return entry(SiEntry("Energy", energy, 2, "J", color))
+    }
+
+    fun farads(capacitance : Double, color : McColor) : DataLabelBuilder {
+        return entry(SiEntry("Capacitance", capacitance, 2, "F", color))
+    }
+
+    fun henry(inductance : Double, color : McColor) : DataLabelBuilder {
+        return entry(SiEntry("Inductance", inductance, 2, "H", color))
+    }
+
+    fun <T> enumerateOn(enumeration: List<T>, editor: (T, DataLabelBuilder) -> Unit) : DataLabelBuilder{
+        for(x in enumeration){
+            editor(x, this)
+        }
+
+        return this
+    }
+}
```

Comment:
I mentioned this before but for the record I don't feel like this is abstract enough and that the colors should be split away from the text generation for this. I want to be able to print out unit types and _then_ apply colors with a function like perhaps `applyColorToText(text: String, color: Color): MinecraftColoredText` or something.

Reply:
Has been addressed in the latest commit. More polishing can be carried out at a later date.